tests/unit/Internal/Primitives/CallableDecoderTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Facile\PhpCodec\Internal\Primitives;
+
+use Eris\Generator as g;
+use Eris\TestTrait;
+use Facile\PhpCodec\Codecs;
+use Tests\Facile\PhpCodec\BaseTestCase;
+use Tests\Facile\PhpCodec\GeneratorUtils;
+
+class CallableDecoderTest extends BaseTestCase
+{
+    use TestTrait;
+
+    public function testLaws(): void
+    {
+        $this
+            ->forAll(
+                g\oneOf(
+                    GeneratorUtils::scalar(),
+                    g\map(
+                        function ($x): callable {
+                            return static function () use ($x) {
+                                return $x;
+                            };
+                        },
+                        GeneratorUtils::scalar()
+                    ),
+                    g\map(
+                        function ($x): callable {
+                            return new class($x) {
+                                private $n;
+
+                                public function __construct($n)
+                                {
+                                    $this->n = $n;
+                                }
+
+                                public function __invoke()
+                                {
+                                    return $this->n;
+                                }
+                            };
+                        },
+                        GeneratorUtils::scalar()
+                    ),
+                    function (): callable {
```

Comment:
I think you're doing this to create a generator that always returns the 'date' callable.
Actually, every parameter of the `forAll` function that not implement the `Generator` interface, is taken as a constant generator.
Hence, the real value generated is an _anonymous function_ that returns the string 'date'.

Instead `g\constant('date')` will do exactly what you want by explicitly calling the constant generator.